Improvements to governance review template

aliok opened the pull request:
https://github.com/cncf/tag-contributor-strategy/pull/479 did not incorporate the improvements suggestions I've made earlier (which is fine).

This PR is based on a v2 of the template that I changed after reviewing Flatcar governance.

There's input from @TheFoxAtWork and @jberkus.

Adding inline comments for the motivation of some of the changes.

Asking reviews from the reviewers of the previous PR: @TheFoxAtWork @geekygirldawn @dzolotusky 
and @jberkus

aliok commented on governance/reviews/template.md:
```diff
@@ -19,35 +33,46 @@ Status: Exemplary|Satisfactory|Mostly Satisfactory|Needs Work
-<!--- The items in the list should be summarized, have a prioritized ordering and are expected to be considered Critical or High, they are usually blockers to project advancement. For each item in this list, a corresponding detailed description should be placed in the Findings table. Note that which items are required depends on the project's maturity level. --->
+<!--- The items in the list should be summarized, have a prioritized ordering and are expected to be considered blockers to project advancement. For each item in this list, a corresponding detailed description should be placed in the Findings table. Note that which items are required depends on the project's maturity level.
+
+Items in the "Governance Findings Table" at the bottom with "Blocker" importance should be reported here as must-fix items.
+
+If there are no must-fix items, do not delete the section but write that there are no must-fix items.
```
I think it is better to classify must-fix items as the ones that are listed as blocker at the end.

Existing severity levels were "Critical/High/Medium/Low" and it was said here that Critical and High are basically blockers. This was confusing for me.

I've made changes to the bottom of the document as well to make this grouping:
- Must-fix: Blocker
- Areas for improvement: High/Medium/Low

jberkus replied:
I'm not sure that the distinction between "High" and "Blocker" would be clear.  Maybe omit "High" entirely, since all High will be Blockers anyway?

And ... why use "Blocker" instead of "MustFix"?

aliok replied:
> Maybe omit "High" entirely, since all High will be Blockers anyway?

My intention is along the same lines. I tried to make distinction between a blocker and a non-blocker.

Let me update with what you've suggested, which is even simpler.

So,
* 3 categories: critical/medium/low
* Critical --> blocker
* Medium/Low --> non-blocker

> And ... why use "Blocker" instead of "MustFix"?

I don't have a strong opinion in this one. I am ok with switching back to MustFix.

aliok replied:
Pushed a new commit for this change.

jberkus replied:
Whatever term we use: "Critical", "Blocker", or "MustFix" should be the same in both the grid and the section header.

aliok commented on governance/reviews/template.md:
```diff
@@ -146,14 +235,27 @@ Areas of potential future development include:
+|:------------- |:----------:|:------------|
+| *Title* | Blocker/High/Medium/Low | *detailed description* |
 
-| Finding Title | Importance | Description | Links | Notes & Impact |
-|:------------- |:----------:|:------------|:------|:---------------|
-| *Title* | Critical/High/Medium/Low | *detailed description* | *relevant links* | *additional notes and explanation of impact if appropriate* |
```
Removed the "Links" and "Notes & Impact" column.
Those were causing duplication of content in title and description columns.

jberkus replied:
So, if there are multiple links for a specific finding, how would that be handled?

aliok replied:
In the description?

In the Flatcar review, I've initially filled the "Links" column. 
Then I found I am adding same links in the description and removed the "Links" column.

Please take a closer look at the Findings Table columns at [WIP review doc for Flatcar](https://github.com/cncf/tag-contributor-strategy/blob/3da188814d42d1bb46e600ebd3b2b7ecb670a75a/governance/assessments/projects/flatcar/2023-08-15.md#governance-findings-table).

aliok replied:
I reverted this change now

jberkus commented on governance/reviews/template.md:
```diff
@@ -62,28 +87,92 @@ Over the next year, the project should work on the following issues to improve i
+Provide the links in the table to the documents/pages that are relevant to the governance area. The links should be to the permanent addresses to snapshots (commits) of the file under evaluation as a point-in-time reference to this review. To ensure consistency, use same commits as links for multiple files within a single repository.
+--->
+
+<!--- 
```
Starting here, this is feeling like this needs to go into an actual Guide To Reviewers doc, instead of inline comments.

aliok replied:
I am ok removing these detailed instructions, if you think a future guide is a better place.

I wouldn't keep the instructions in 2 places as they will become out-of-sync eventually.

aliok replied:
I removed the detailed comments now.

jberkus commented on governance/reviews/template.md:
```diff
@@ -62,28 +87,92 @@ Over the next year, the project should work on the following issues to improve i
+
+--->
+
+| Governance Area | Coverage | Finding Notes |
```
The reason for the LINKS column is that in some cases, projects may have multiple documents that cover a particular area.  As such, making the area name a link is not sufficient.

aliok replied:
This makes sense. Reverting the columns in this table.

aliok replied:
Reverted

jberkus commented on governance/reviews/template.md:
```diff
@@ -111,17 +200,17 @@ Governance Drift can occur when the executed and observable governance of a proj
 The project [does/does] not experience governance drift as indicated by...
 
-#### Ownership
+### Ownership
```
no, this was on purpose.  Ownership is linked to Drift, rather than being separate from it

aliok replied:
I think you mean Ownership is linked to "Operation", but I got it.

jberkus commented:
OK, requested some discussion on some of these changes.

The big thing is that I think we've crossed over the line from "Inline comments" into "Needs a reviewer guide doc".

aliok commented:
@jberkus @geekygirldawn 

* I removed the detailed instructions
* I reverted the changes I made to Governance Finding Table structure

Can you have a new look please?

geekygirldawn commented on governance/reviews/template.md:
```diff
@@ -38,16 +60,20 @@ The following aspects of governance are exemplary, and can be referenced as exam
 
 <!--- Narrative describing the governance type of the project, some general information about its leadership, and the project's general status and maturity. If the project has any unusual aspects to its governance, describe them here.  Link to the project's existing documents where applicable. --->
 
+<!--- You will most likely need the number of substantial contributors in the project and write it down along with their employer affiliations. If project is already under CNCF, use Devstats; if not, manually check the repositories. --->
```
I'm not sure about this. There isn't a requirement to document substantial contributors, but there is a requirement for having maintainers from at least 2 companies, so I focus on the maintainers. I also usually start by getting maintainer affiliations from the governance docs, since most of them are listed there. I think it's harder to pull this out of devstats for individual contributors. Also, I don't think we've done any governance assessments where the project isn't already under the CNCF. I guess maybe it could happen if we're asked to assess a project that is being proposed to come in as an incubating project?

aliok replied:
> Also, I don't think we've done any governance assessments where the project isn't already under the CNCF.
I've done it for Flatcar: https://github.com/cncf/tag-contributor-strategy/blob/main/governance/assessments/projects/flatcar/2023-08-15.md

> I guess maybe it could happen if we're asked to assess a project that is being proposed to come in as an incubating project?

I've only done a single review so far. After my single review, I thought this is a general thing to do and wanted to document what I've done for any future reviewers. Thanks catching this!

I will update this instruction to make it only applies for projects that want to apply to CNCF.

aliok replied:
Removed!

geekygirldawn reviewed (commented):
This is great! Overall, I think what you've added is really helpful. I only left a comment about one that seemed a little confusing to me (or different from what I've experienced).

aliok commented:
@jberkus @geekygirldawn Would you like to merge this one?

aliok commented:
The wording of the "Critical" and "Blocker" should be reduced into 1 wording

geekygirldawn reviewed (approved):
Approved pending the fix to be consistent between ‘critical’ and ‘blocker’ throughout the document.

aliok commented:
> Approved pending the fix to be consistent between ‘critical’ and ‘blocker’ throughout the document.

This is done now.

cc @geekygirldawn @jberkus
